py_gtfs_rt_ingestion/dev_test_setup.py

```diff
@@ -0,0 +1,298 @@
+
+
+def get_del_obj_list(bucket: str, uri_root: str) -> list[str]:
+    """
+    Return list of objects in bucket, removing uri_root object from return list
+    """
+    print(f"Checking for objects to delete in {uri_root}...")
+    files_to_delete = []
+    for (uri, _) in file_list_from_s3(bucket, LAMP_PREFIX):
+        files_to_delete.append(uri)
+    # Remove uri_root from list so that root directory is not deleted.
+    files_to_delete.remove(uri_root)
+    return sorted(files_to_delete, reverse=True)
+
+
+def clear_dev_buckets(args: SetupArgs) -> None:
+    """
+    Clear all development buckets of objects
+    """
+    bucket_list = (
+        DEV_INGEST_BUCKET,
+        DEV_EXPORT_BUCKET,
+        DEV_ARCHIVE_BUCKET,
+        DEV_ERROR_BUCKET,
+    )
+
+    for bucket in bucket_list:
+        action = None
+        uri_root = f"s3://{os.path.join(bucket, LAMP_PREFIX)}"
+        files_to_delete = get_del_obj_list(bucket, uri_root)
+        if len(files_to_delete) == 0:
+            print("No objects found... skipping bucket.")
+        else:
+            while action not in ("n", "no"):
+                # Print list of bucket objects
+                if action in ("list", "ls"):
+                    print(f"/{'*' * 50}/")
+                    for uri in files_to_delete:
+                        print(f"{uri.replace(uri_root, '')}")
+                    print(f"/{'*' * 50}/")
+                # Proceed with object deletion
+                elif action in ("y", "yes") or args.force_delete:
+                    delete_count = make_del_jobs(files_to_delete, bucket)
+                    # If not all objects deleted, retry
+                    if delete_count < len(files_to_delete):
+                        print(
+                            f"Only {delete_count:,} of {len(files_to_delete):,}"
+                            " deleted... will retry."
+                        )
+                        files_to_delete = get_del_obj_list(bucket, uri_root)
+                    else:
+                        print(f"All {len(files_to_delete)} deleted")
+                        break
+
+                print(
+                    f"{len(files_to_delete):,} objects found, delete? "
+                    "\n"
+                    "yes(y) / no(n) / list(ls)"
+                )
+                action = input()
+
+
+def copy_obj(prefix: str, num_to_copy: int) -> int:
+    """
+    Copy random objects from prefix to dev Ingest bucket
+    """
+    count_objs_to_pull = max(num_to_copy * 10, 1_000)
+    uri_copy_set = set()
+    src_uri_root = f"s3://{os.path.join(GTFS_BUCKET, prefix)}"
+    skip_uri = "https_mbta_busloc_s3.s3.amazonaws.com_prod_TripUpdates_enhanced"
+    print(
+        f"Pulling list of {count_objs_to_pull:,} objects from "
+        f"{src_uri_root} for random sample..."
+    )
+    try:
+        for (uri, size) in file_list_from_s3(GTFS_BUCKET, prefix):
+            # Skip busloc TripUpdates because of S3 permission issues
```

Comment:
what was the deal with this again?

they have different permissions set even though they are in the same bucket?

Reply:
Yup, it's a known issue https://app.asana.com/0/1200152089785426/1200192016253762/f